src/ImageSharp/Formats/ImageExtesions.Save.cs

```diff
@@ -0,0 +1,694 @@
+        /// <param name="path">The file path to save the image to.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        public static void SaveAsBmp(this Image source, string path) => SaveAsBmp(source, path, null);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, string path) => SaveAsBmpAsync(source, path, null);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, string path, CancellationToken cancellationToken)
+            => SaveAsBmpAsync(source, path, null, cancellationToken);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <param name="encoder">The encoder to save the image with.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        public static void SaveAsBmp(this Image source, string path, BmpEncoder encoder) =>
+            source.Save(
+                path,
+                encoder ?? source.GetConfiguration().ImageFormatsManager.FindEncoder(BmpFormat.Instance));
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <param name="encoder">The encoder to save the image with.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, string path, BmpEncoder encoder)
```

Comment:
You mean like this I guess:
```C#
public static Task  `SaveAsBmpAsync(this Image source, string path, BmpEncoder encoder, CancellationToken ct = default)
```

Back in the days, we agreed on a guideline with @JimBobSquarePants that discourages default args. This would be a good example for the reason: introducing the variant with default args, and dropping `SaveAsBmpAsync(this Image source, string path, BmpEncoder encoder)` will break on binary level. This is a very well visible API, and we are already RC, so we want to avoid this for sure.

Reply:
Yep, can't break that.

Reply:
don't see why we would shy away from breaking it yet...we only just released the async apis theres not likely to be much prior art on it...and binary braking changes should be an issue yet not until a non-preview/rc release. it isn't a source brake

Reply:
> we only just released the async apis theres not likely to be much prior art on it

This makes sense. It's a limited break between RC3 -> 1.0. I'm starting to like the idea. @JimBobSquarePants thoughts?

Reply:
Ok then. If it's convention, it's convention. Carry on 👍